spring-cloud-gcp-data-datastore/src/main/java/org/springframework/cloud/gcp/data/datastore/repository/query/PartTreeDatastoreQuery.java

```diff
@@ -347,6 +343,17 @@ private void applySelectWithFilter(Object[] parameters, Builder builder) {
 						: filters[0]);
 	}
 
+	private Value convertParam(DatastorePersistentProperty persistentProperty, Object val) {
+		if (persistentProperty.isAssociation()
```

Comment:
why is `persistentProperty.isAssociation()` important?

Reply:
that means it's a reference, we want to convert objects to keys only if that is meaningful